src/etc/inc/interfaces.inc

```diff
@@ -3196,7 +3196,35 @@ function interface_dhcpv6_configure($interface = 'wan', $wancfg)
     unset($dhcp6cconf);
 
     $dhcp6cscript = "#!/bin/sh\n";
+    $dhcp6cscript .= "dmips=\${new_domain_name_servers}\n";
+    $dhcp6cscript .= "dmnames=\${new_domain_name}\n";
+    $dhcp6cscript .= "case \$REASON in\n";
+    $dhcp6cscript .= "REQUEST)\n";
     $dhcp6cscript .= "/usr/local/opnsense/service/configd_ctl.py interface newipv6 {$wanif}\n";
+    if (isset($wancfg['adv_dhcp6_debug'])) {
+        $dhcp6cscript .= "/usr/bin/logger -t dhcp6c \"dhcp6c REQUEST on {$wanif} - running newipv6\"\n";
+    }
+    $dhcp6cscript .= ";;\n";
+    $dhcp6cscript .= "REBIND)\n";
+    if (!empty($wancfg['adv_dhcp6_debug'])) {
+        $dhcp6cscript .= "/usr/bin/logger -t dhcp6c \"dhcp6c rebind on {$wanif}\"\n";
+    }
+    $dhcp6cscript .= ";;\n";
+    if (!empty($wancfg['dhcp6norelease'])) {
```

Comment:
what's the logic here? it flips EXIT to RELEASE but wouldn't it make more sense to omit the block when dhcp6norelease is set?

Reply:
No, even if dhcp6c exits with the no release option, it still exits and removes the leases, but the reason is different. Makes sense to catch it either way. You could just have both in there, but I was trying to reduce code when I wrote the original for p*****e a while back.

Reply:
Ah, so in a nutshell we want to do updates under more restrictive circumstances?

Reply:
Yes, it also creates a framework for what may happen in future.

I am still working my way through this area of code, it was ripe for issues with dhcp6c, under p*****e and the pitfalls are here too, well they would be, :) 

I fairly easily managed to get two instances of dhpc6c running, I and my testers spent many happy hours/days resolving the issues with dhcp6c, however I assume that this section is up for a complete re-write? 

If you want me to, I can easily recreate the same things that were done on p******, I have no qualms about using code I originally wrote.

Reply:
Ah, I misunderstood your query about exit or release, hmm, I'm trying to remember why that was done, because  its easier to use an or like in the case of info RENEW|INFO. Looking at it now I can see no reason at all to do that, either will result in the same action..

I think it was just down to testing, I have just looked at some old interfaces.inc files and there were just more debug messages while we ( my testers ) were working out what would happen with various responses. 

I'll clean that bit up too,

Reply:
Alright, thanks for explaining. :) In that case "\$REASON" could be used to condense the debug messages. I'll take a closer look next week.

Reply:
Yes. :) 

I thought of just doing a single if(dbeg-blah) log reason is {$REASON}
but there was a reason I didn't, now I cannot remember what it was.

Having issues trying to get this dhcp6 VLAN prio to work, but that's another topic.

Reply:
hehe, if it comes back to memory just let me know

I've added a side note to that topic, not sure if useful https://forum.opnsense.org/index.php?topic=6714.0